pkg/kine/server/update.go

```diff
@@ -46,26 +46,26 @@ func (l *LimitedServer) update(ctx context.Context, rev int64, key string, value
 
 	if rev == 0 {
 		rev, err = l.backend.Create(ctx, key, value, lease)
-		ok = true
-
-		span.SetAttributes(
-			attribute.Int64("revision", rev),
-			attribute.Bool("ok", ok),
-		)
+		if err == ErrKeyExists {
+			updated = false
```

Comment:
So this would mean there is a case where the update request is asked to update a rev 0, the create returns rev 0 and a nil error? I don't think it is possible or when does this happen?

Reply:
A create is a transaction like this:

```
if mod_revision(key) == 0
then put(key, value)
else do nothing
```

In the `else` branch we are returning a `ResponseRange` (and performing a `get`) instead of nothing.

In the code the path is:
 - we call `l.backend.Create` and it returns `0, ErrKeyExists` (so `rev = 0` here) as the key is already there;
 - then we do `if err == ErrKeyExists { updated = false; err = nil }` (line 49, 50, 51) which is correct as it is expected by the transaction code that the row could already be there.
 
 However, in this case we reach the `else` branch on line `78` which performs a `get` and returns the prev value. This was not required by this transaction and should really not be done.
 
Reading this again, I don't think adding a simple `if rev != 0` is the solution as that would prevent the `update` path (which does require the prev value) from getting the row. Maybe we should not overwrite `rev`? (it is in general a bad idea to overwrite parameters...)

Reply:
I think the easiest is to return the response right there that will be easier to follow than adding another case in the later logic. Have a look please and let me know what you think :)